tools/lint/run_clang_tidy.sh

```
@@ -0,0 +1,22 @@
+
+cmake -S . -B "${build_dir}" -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+cmake --build "${build_dir}"
+
+sources=$(git ls-files '*.c')
+if [ -z "${sources}" ]; then
+  echo "clang-tidy: no C sources to analyze"
+  exit 0
+fi
+
+if [[ "$(uname)" == "Darwin" ]]; then
+  export SDKROOT="$(xcrun --show-sdk-path)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)